sdk/include/ddk/ntstrsafe.h

```diff
@@ -2825,6 +2826,45 @@ RtlStringLengthWorkerW(
     return Status;
 }
 
+NTSTRSAFEAPI 
+RtlUnicodeStringPrintf(PUNICODE_STRING DestinationString,
+	NTSTRSAFE_PCWSTR pszFormat, ...)
+{
+	if (DestinationString == NULL || pszFormat == NULL)
+		return STATUS_INVALID_PARAMETER;
+
+	if (DestinationString->Buffer == NULL || DestinationString->MaximumLength == 0)
+		return STATUS_INVALID_PARAMETER;
+	
+	size_t cchLength = 0;
+	NTSTATUS nt = RtlStringLengthWorkerW(DestinationString->Length, NTSTRSAFE_UNICODE_STRING_MAX_CCH, &cchLength);
+	ASSERT(NT_SUCCESS(nt));
+	
+	/* Don't allow inconsistency between actual length and UNICODE_STRING.Length field, 
+	 * or "full buffer" (MaximumLength == Length)
+	 */
+	
+	if ( ((DestinationString->Length == 0) && (cchLength != 0)) ||
+		 ((DestinationString->Length == DestinationString->MaximumLength)))
+		return STATUS_INVALID_PARAMETER;
+
+	return RtlStringVPrintfWorkerW(DestinationString->Buffer, DestinationString->MaximumLength, pszFormat, argList);
+}
+
+NTSTRSAFEAPI 
+RtlUnicodeStringPrintfEx(PUNICODE_STRING DestinationString,
+	PUNICODE_STRING RemainingString,
+	DWORD dwFlags,
+	NTSTRSAFE_PCWSTR pszFormat, ...)
+{
+	/* Verify flags */
+
+	if (dwFlags == STRSAFE_FILL_BEHIND)
+	{
+
+	}
```

Comment:
I think that here you can just call RtlStringPrintfExW with the correct parameters (the UNICODE_STRING already provide the needed buffer pointers and their buffer lenghts).